doc/developer/design/20231110_aws_connections.md

```diff
@@ -107,16 +112,50 @@ When creating the connection, users should specify `WITH (VALIDATE = false)` in
 This is because we won't be able to validate the connection unless the trust policy has
 been added on the user's end.
 
+#### `mz_aws_connections` table
+
+We'll add a new `mz_aws_connections` table with the following definition:
+
+```sql
+CREATE TABLE mz_internal.mz_aws_connections (
+    -- The ID of the connection in `mz_connections`.
+    id text NOT NULL,
+    -- The value of the `ENDPOINT` option, if specified. `NULL` otherwise.
+    endpoint text,
+    -- The value of the `REGION` option, if specified. `NULL` otherwise.
+    region text,
+    -- The value of the `AWS ACCESS KEY ID` option, if specified. `NULL` otherwise.
+    access_key_id text,
+    -- The value of the `ASSUME ROLE ARN` option, if specified. `NULL` otherwise.
+    assume_role_arn text,
+    -- The value of the `ASSUME ROLE SESSION ARN` option, if specified. `NULL` otherwise.
+    assume_role_session_name text,
+    -- The AWS IAM principal Materialize will use to assume the role, if
+    -- the connection configured `ASSUME ROLE ARN`. `NULL` otherwise.
+    principal text,
+    -- The external ID Materialize will use to assume the role, if
+    -- the connection configured `ASSUME ROLE ARN`. `NULL` otherwise.
+    external_id text,
+    -- A fully rendered AWS IAM trust policy that should be installed on the
+    -- role specified by `ASSUME ROLE ARN` to grant Materialize permission to
+    -- assume the role. `NULL` if `ASSUME ROLE ARN` is not specified.
+    trust_policy text
+)
+```
+
 #### External ID and principal
 We already have external ID prefix provided in the catalog state.
 The connection ID will be appended to the external ID prefix to get the complete
 External ID and then stored in the new `mz_aws_connections` table for that connection ID.
 
-For principal, we'll create one global role named something like
-`arn:aws:iam::<account_id>:role/MaterializeCloudConnectionAssumeRole`
+For principal, we'll create one global role named
```

Comment:
@jubrad @moulimukherjee I felt like the old name was too verbose. We've dropped "Cloud" from our public messaging. `AssumeRole` is implied—roles are always things you assume! So that leaves `MaterializeConnection` or `Materialize`. I prefer the shorter name myself. From the customer's perspective, just `Materialize` captures everything they care about, I think—it's the role Materialize will be connecting as!

Reply:
Yeah this seems good... I maybe `MaterializeConnection` would be best. I think the goal is:
1. Ensure end users get the feeling that is specific enough
2. Ensure there's no confusion about the purpose of this role when assuming it or seeing it in audit logs on either side.

Also... not all roles are meant for assuming other roles so AssumeRole is not really implied, that being said it is far from short.

Reply:
> Also... not all roles are meant for assuming other roles so AssumeRole is not really implied, that being said it is far from short

Oh, I see—the idea was that `AssumeRole` meant "we use this role to _assume other roles_"! I had read it as "we assume this role for connections." It's too bad there isn't a term of art like "jump role" like there is for "jump host."

Reply:
Just throwing out more options:
- `MaterializeIntermediate` | `MaterializeIntermediateRole`
- `MaterializeTunnel`
- `MaterializeExternal` | `MaterializeExternalRole`
- `MaterializeAssumeRole` | `AssumeRoleMaterialize`

Reply:
I definitely want a more descriptive name. We may also be forced to include some additional information in the name, at least in dev, since they share an AWS account.

My vote goes to `MaterializeRoleAssumer` or `MaterializeExternalRoleAssumer`.

Reply:
> We may also be forced to include some additional information in the name, at least in dev, since they share an AWS account.

Yep, it's orthogonal to the questions here, but we'll need to suffix the stack name in development.

> I definitely want a more descriptive name. We may also be forced to include some additional information in the name, at least in dev, since they share an AWS account.
> 
> My vote goes to `MaterializeRoleAssumer` or `MaterializeExternalRoleAssumer`. If we're forced to include additional info, we can prefix or suffix the stack name. I think IAM is global, so I don't think we need to include the region.

Thinking about it from the end user's perspective, though, what does `RoleAssumer` or `ExternalRoleAssumer` make clear? IMO the role should be named for _who_ it represents, and the clearest "who" from the user's perspective is, I think, just `Materialize`. It looks quite clear in the trust policy:

```json
{
   "Version": "2012-10-17",
   "Statement": [
     {
       "Effect": "Allow",
       "Principal": {
         "AWS": "arn:aws:iam::001234567890:role/Materialize"
      },
      "Action": "sts:AssumeRole",
      "Condition": {
        "StringEquals": {
          "sts:ExternalId": "<>"
        }
      }
    }
  ]
 }
```

vs:

```json
{
   "Version": "2012-10-17",
   "Statement": [
     {
       "Effect": "Allow",
       "Principal": {
         "AWS": "arn:aws:iam::001234567890:role/MaterializeExternalRoleAssumer"
      },
      "Action": "sts:AssumeRole",
      "Condition": {
        "StringEquals": {
          "sts:ExternalId": "<>"
        }
      }
    }
  ]
 }
```

Reply:
I was thinking about this last night some as well and I'm definitely growing more in favor of brevity. The criteria for us is that it's unique and we wont' conflate it with anything else. The criteria for the user is that it identifies the purpose and gives them good vibes.

I don't think we should do `MaterializeExternalRoleAssumer`, we don't want them to waste time contemplating our implementation.

I also have a concern that simply naming it just `Materialize` might go in the opposite direction and throw up a redflag for sec, in that it may appear that it is not unique to connections and potentially broadly used by materialize. Again they may potentially waste time contemplating our implementation.

I'm leaning `MaterializeConnection`... it's specific enough to show purpose without implying implementation. It makes sense for us and for customers.

Reply:
`MaterializeConnection` is fine with me.

Reply:
`MaterializeConnection` sgtm!